curvefs/src/client/metric/client_metric.h

```diff
@@ -103,6 +103,84 @@ struct MetaServerClientMetric {
           createPartition(prefix, "createPartition") {}
 };
 
+struct LatencyUpdater {
+    explicit LatencyUpdater(bvar::LatencyRecorder* recorder)
+        : recorder(recorder) {
+        timer.start();
+    }
+
+    ~LatencyUpdater() {
+        timer.stop();
+        (*recorder) << timer.u_elapsed();
+    }
+
+    bvar::LatencyRecorder* recorder;
+    butil::Timer timer;
+};
+
+struct OpMetric : public InterfaceMetric {
```

Comment:
I think there're only four types of metric that need to record

1. qps
2. latency
3. eps or error count
4. inflight count

so `OpMetric` is not suitable for this work, because `InterfaceMetric` has more metrics.

and, another tip is `LatencyRecorder` is an aggregate metric, it already supports qps by default.  https://github.com/apache/incubator-brpc/blob/0148f46e3b8b080808d031ba425555392dfac894/src/bvar/latency_recorder.h#L49

so, please redesign this class.

Reply:
fixed